Web Inspector: Console: add an internal property for the `target` of a `WeakRef`

dcrousso opened the pull request:
#### c3366d5b4fd8c4e40e4cded5ac621dca518d5813
<pre>
Web Inspector: Console: add an internal property for the `target` of a `WeakRef`
<a href="https://bugs.webkit.org/show_bug.cgi?id=256710">https://bugs.webkit.org/show_bug.cgi?id=256710</a>

Reviewed by Patrick Angle.

Add special handling for `WeakRef` to make it easier for developers to at-a-glance know what the `WeakRef` is for (i.e. what&apos;s the `target`).

* Source/JavaScriptCore/inspector/InjectedScriptSource.js:
(InjectedScript.prototype._forEachPropertyDescriptor):
(RemoteObject):
(RemoteObject.describe):
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.h:
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
(Inspector::JSInjectedScriptHost::subtype):
(Inspector::JSInjectedScriptHost::getInternalProperties):
(Inspector::JSInjectedScriptHost::weakRefTargetValue): Added.
* Source/JavaScriptCore/inspector/JSInjectedScriptHostPrototype.cpp:
(Inspector::JSInjectedScriptHostPrototype::finishCreation):
(Inspector::jsInjectedScriptHostPrototypeFunctionWeakRefTargetValue): Added.
* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._formatParameter):

* LayoutTests/inspector/model/remote-object/weakref.html: Added.
* LayoutTests/inspector/model/remote-object/weakref-expected.txt: Added.

* LayoutTests/inspector/model/remote-object/proxy.html: Added.
* LayoutTests/inspector/model/remote-object/proxy-expected.txt: Added.

Drive-by: Add tests for `Proxy` accidentally removed in 217300@main.
Canonical link: <a href="https://commits.webkit.org/264171@main">https://commits.webkit.org/264171@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/83c7ba781c054ed756b1a50801425f17d73ac2fa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6913 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7143 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7326 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8512 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7159 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7082 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10061 "Failed to checkout and rebase branch from PR 13813") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7035 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/7709 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6368 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8612 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/5089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6278 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/5837 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6364 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9183 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6488 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6845 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7053 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6221 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/6190 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1650 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1641 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10406 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/7244 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6599 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1785 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

patrickangle commented on LayoutTests/inspector/model/remote-object/weakref.html:
```diff
@@ -0,0 +1,24 @@
+function test()
+{
+    let steps = [
+        {expression: `new WeakRef({x:1, y:1})`},
```
It looks like you implemented support for resolving the target of a WeakRef of a WeakRef. Can we go ahead and test that?

dcrousso committed:
Web Inspector: Console: add an internal property for the `target` of a `WeakRef`
https://bugs.webkit.org/show_bug.cgi?id=256710

Reviewed by Patrick Angle.

Add special handling for `WeakRef` to make it easier for developers to at-a-glance know what the `WeakRef` is for (i.e. what's the `target`).

* Source/JavaScriptCore/inspector/InjectedScriptSource.js:
(InjectedScript.prototype._forEachPropertyDescriptor):
(RemoteObject):
(RemoteObject.describe):
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.h:
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
(Inspector::JSInjectedScriptHost::subtype):
(Inspector::JSInjectedScriptHost::getInternalProperties):
(Inspector::JSInjectedScriptHost::weakRefTargetValue): Added.
* Source/JavaScriptCore/inspector/JSInjectedScriptHostPrototype.cpp:
(Inspector::JSInjectedScriptHostPrototype::finishCreation):
(Inspector::jsInjectedScriptHostPrototypeFunctionWeakRefTargetValue): Added.
* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._formatParameter):

* LayoutTests/inspector/model/remote-object/weakref.html: Added.
* LayoutTests/inspector/model/remote-object/weakref-expected.txt: Added.

* LayoutTests/inspector/model/remote-object/proxy.html: Added.
* LayoutTests/inspector/model/remote-object/proxy-expected.txt: Added.

Drive-by: Add tests for `Proxy` accidentally removed in 217300@main.
Canonical link: https://commits.webkit.org/264171@main
